lib/developer_portal/lib/liquid/template/view.rb

```diff
@@ -29,7 +29,10 @@ def self.from(record, path, cms)
 
         ::Rails.logger.debug { "Liquid::Template::View for #{path.inspect} - #{record.inspect}" }
 
-        content = record.content(cms.try(:render_draft_content?))
+        # Default to "" to avoid an issue with strict locals, where
+        # `self.source.sub!(STRICT_LOCALS_REGEX, "")` fails with "can't modify frozen String", because
+        # `self.source` is `nil.to_s`, which is frozen
+        content = record.content(cms.try(:render_draft_content?)) || ""
```

Comment:
I see that the file does not use frozen string literals, ideally it should.

Either way, to make it clear that this empty string is supposed to be unfrozen, better make it like this
```suggestion
        content = record.content(cms.try(:render_draft_content?)) || String.new
```